groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -320,14 +327,18 @@ template <class TYPE, class ANY_TYPE>
 struct Optional_ConvertsFromStdOptional
 : bsl::integral_constant<
       bool,
-      Optional_ConvertsFrom<TYPE, std::optional<ANY_TYPE> >::value> {
+      Optional_ConvertsFrom<TYPE,
+                            std::optional<typename Optional_RemoveCVRef<
+                                ANY_TYPE>::type> >::value> {
```

Comment:
it actually wasn't at all needed here because it was in the outer assigns/convert_FromOptional.  Now that the outer traits are gone, I've moved the stripping of cvref into the st/bsl specific traits.